File: src/main/java/com/algorithm/datastructure/boj/queue/BOJ_10845.java
package com.algorithm.datastructure.boj.queue;

import java.io.*;
import java.util.*;

public class BOJ_10845 {

    public static void main(String[] args) throws IOException{

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int cmdCnt = Integer.parseInt(br.readLine());
        String[] queue = new String[10001];
        int front = 0;
        int back = 0;


        for (int i = 0; i < cmdCnt; i++) {
            String cmdTxt = br.readLine();
            StringTokenizer st = new StringTokenizer(cmdTxt);
            String cmd = st.nextToken();

            switch(cmd){
                case "push":
                    queue[back++] = st.nextToken();
                    break;
                case "front":
                    String ftemp = front != back ? queue[front] : "-1";
                    System.out.println(ftemp);
                    break;
                case "back":
                    String btemp = front != back ? queue[back-1] : "-1";
                    System.out.println(btemp);
                    break;
                case "size":
                    int size = back - front;
                    System.out.println(size);
                    break;
                case "empty":
                    int isEmpty = front == back ? 1 : 0;
                    System.out.println(isEmpty);
                    break;
                case "pop":
                    if(front != back) {
                        System.out.println(queue[front++]);
                    }else{
                        System.out.println("-1");
                    }
                    break;
                default:
                    break;
            }


        }

    }
}

File: src/main/cpp/algorithm/basic/BOJ_10845.cpp
#include <iostream>
#include <string>
#include <vector>
using namespace std;

int main(void){
    int n;
    cin >> n;
    vector<int> q;
    for(int i = 0; i < n; i++){
        string s;
        cin >> s;

        if(s=="push"){
            int n;
            cin >> n;
            q.push_back(n);
        }else if(s=="pop"){
            if(q.empty()){
                cout << -1 << "\n";
            }else{
                cout << q[0] << "\n";
                q.erase(q.begin());
            }
        }else if(s=="size"){
            cout << q.size() << "\n";
        }else if(s=="empty"){
            if(q.empty()) cout << 1 <<"\n";
            else cout << 0 << "\n";
        }else if(s=="front"){
            if(q.empty()) cout << -1 << "\n";
            else cout << q[0] << "\n";
        }else if(s=="back"){
            if(q.empty()) cout << -1 << "\n";
            else cout << q[q.size()-1] << "\n"; 
        }
    }
    return 0;
}